test/integration/graphql_test.lua

```diff
@@ -1639,3 +1639,53 @@ function g.test_schema_input_arg_described_with_kind_variable_pass()
     local _, errors = check_request(query, query_schema, nil, nil, { variables = variables })
     t.assert_equals(errors, nil)
 end
+
+function g.test_arguments_default_values()
+    local function callback(_, _)
+        return nil
+    end
+
+    local mutation_schema = {
+        ['test_mutation'] = {
+            kind = types.string.nonNull,
+            arguments = {
+                mutation_arg = types.string,
+                mutation_arg_defaults = {
+                    kind = types.inputObject({
+                        name = 'test_input_object',
+                        fields = {
+                            input_object_arg_defaults = {
+                                kind = types.string,
+                                defaultValue = 'input object argument default value'
+                            },
+                            input_object_arg = types.string,
+                            nested_enum_arg_defaults = {
+                                kind = types.enum({
+                                    schema = schema,
+                                    name = 'mode',
+                                    values = {
+                                        read = 'read',
+                                        write = 'write',
+                                    },
+                                }),
+                                defaultValue = 'write',
+                            }
+                        },
+                        kind = types.string,
+                    }),
+                },
+            },
+            resolve = callback,
+        }
+    }
+
+    local data, errors = check_request(introspection.query, nil, mutation_schema)
```

Comment:
I'm still confused about how graphql works, so I apologize in advance if my question is stupid.

Does this fix affect only introspection query response or it also fixes some behavior for other queries?

Reply:
This particular fix solves the problem that defaultValues is not propagated to schema and also to final introspection json.

Reply:
It's just would be more useful (and honest) to test the change with executing some mutation request and check on the receiving side (somewhere in callback or callback result) that default fields was set up if user haven't provided the values.